packages/validation/frontend/sirius-components-validation/src/useValidationViewSubscription.tsx

```diff
@@ -60,27 +62,33 @@ export const useValidationViewSubscription = (
 
   const onComplete = () => setState((prevState) => ({ ...prevState, complete: true }));
 
-  const onData = ({}: OnDataOptions<GQLValidationEventSubscription>) =>
-    setState((prevState) => ({ ...prevState, complete: false }));
+  const onData = ({ data }: OnDataOptions<GQLValidationEventSubscription>) =>
+    flushSync(() => {
+      if (data.data) {
+        const { validationEvent } = data.data;
+        if (validationEvent) {
+          setState((prevState) => ({ ...prevState, payload: validationEvent, complete: false }));
+        }
+      }
+    });
+
+  const { addErrorMessage } = useMultiToast();
+  const onError = ({ message }: ApolloError) => {
+    addErrorMessage(message);
+  };
 
-  const { data, error, loading } = useSubscription<GQLValidationEventSubscription, GQLValidationEventVariables>(
+  const { data, loading } = useSubscription<GQLValidationEventSubscription, GQLValidationEventVariables>(
```

Comment:
Should we read (and use below) the `data` returned here, or the one set in the state by `onData`?

Reply:
indeed, thanks